include/zephyr/drivers/gpio.h

```diff
@@ -555,6 +555,20 @@ __subsystem struct gpio_driver_api {
  * @endcond
  */
 
+/**
+ * @brief Validate that GPIO port is ready.
+ *
+ * @param spec GPIO specification from devicetree
+ *
+ * @retval true if the GPIO spec is ready for use.
+ * @retval false if the GPIO spec is not ready for use.
+ */
+static inline bool gpio_is_ready(const struct gpio_dt_spec *spec)
```

Comment:
I think this should be named gpio_is_ready_dt() for consistency with other helpers that take a gpio-dt_spec.

Reply:
But then we won't follow the pattern which is already done for the SPI which is:

`static inline bool spi_is_ready(const struct spi_dt_spec *spec)`

Should we rename that as well?

Reply:
> But then we won't follow the pattern which is already done for the SPI which is:
>
> static inline bool spi_is_ready(const struct spi_dt_spec *spec)



Ugh, I wasn't aware of that function. Looks like it was introduced in dcc04d9abda058ed7fa6561b7fecbcc62c6e3d3f. @JordanYates would you agree that spi_is_ready() would have been better named spi_is_ready_dt(), for consistency with all the other _dt suffixed functions that operate on foo_dt_spec structures? (Asking since git says you were the original author of that function).

Reply:
I see now that I approved the PR adding spi_is_ready; I must have missed this at the time...

Reply:
> Ugh, I wasn't aware of that function. Looks like it was introduced in https://github.com/zephyrproject-rtos/zephyr/commit/dcc04d9abda058ed7fa6561b7fecbcc62c6e3d3f. @JordanYates would you agree that spi_is_ready() would have been better named spi_is_ready_dt(), for consistency with all the other _dt suffixed functions that operate on foo_dt_spec structures? (Asking since git says you were the original author of that function).

I agree that `spi_is_ready_dt` probably would be better for consistency. From memory I added that function towards the start of `_dt` usage when there were fewer examples. We could certainly add `spi_is_ready_dt` now and mark the original as deprecated

Reply:
I've renamed that to `gpio_is_ready_dt`. I'm gonna fix the SPI function as well next week.

Reply:
@mbolivar-nordic @JordanYates SPI part has already been done. https://github.com/zephyrproject-rtos/zephyr/pull/52770.